client/my-sites/customer-home/main.jsx

```diff
@@ -56,6 +59,15 @@ const Home = ( {
 		);
 	}
 
+	if ( 'purchase-success' === noticeType && layout ) {
+		const successMessage = translate( 'Your purchase has been completed!' );
+		reduxDispatch(
```

Comment:
Hm... one more thing I was thinking of. This might be fine but I wonder if this should happen on every render, or only when `noticeType` changes? I'm thinking that if it happens on every render as it is currently that it might recreate the notice even if it's dismissed. I'm not sure.

Anyway, if you'd like to do this only when `noticeType` changes, something like this should work:

```js
useEffect(() => {
  if (! layout || noticeType !== 'purchase-success') {
    return;
  }
  const successMessage = translate( /* ... */ );
  reduxDispatch( successNotice( /* ... */ ) );
}, [noticeType, layout, reduxDispatch, translate]);
```

Reply:
Good point! I've added a fix for this in [6bccc79](https://github.com/Automattic/wp-calypso/pull/49612/commits/6bccc794ba504cc70032d9d52eec60625cc231fa).